pkg/sdn/plugin/eventqueue.go

```diff
@@ -0,0 +1,140 @@
+package plugin
+
+import (
+	"fmt"
+
+	"k8s.io/kubernetes/pkg/client/cache"
+)
+
+func NewEventQueue(keyFunc cache.KeyFunc) *cache.DeltaFIFO {
+	return cache.NewDeltaFIFO(
+		keyFunc,
+		cache.DeltaCompressorFunc(func(d cache.Deltas) cache.Deltas {
+			return deltaCompressor(d, keyFunc)
+		}),
+		cache.NewStore(keyFunc),
+	)
+}
+
+// Describes the effect of processing a cache event on the delta list.
+type cacheEventEffect string
+
+const (
+	// The cache event should result in an add
+	cacheEventEffectAdd cacheEventEffect = "ADD"
```

Comment:
The naming here is confusing. I think the internal delta list management needs to use different terms than the external event queue (to make it clear that this is about adding an item to the list of notifications, and not necessarily about an item being added to the set the caller is monitoring).